### PR TITLE
ENH - GH:17160, DataFrame.pivot accepts a list of values

### LINKED ISSUE #17160: Make `DataFrame.pivot` accepts a list of column names as values
#### Problem description

Currently `DataFrame.pivot` accepts either `None` or a column name for the option `values` but not a list of column names.
```python
# this works
df.pivot(values = None)
df.pivot(values = 'column_name')
```
```python
# this doesn't work
df.pivot(values = ['cn1', 'cn2'])
```
 It is not convenient if I want to create a pivot table for multiple columns at the same time. Of course, we can pass `None` to `values` and then select columns. But this is not as convenient.

### PR DESCRIPTION
**Pull request by BBQHarvey**
- [x] closes #17160
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by BBQHarvey** [comment]
I can't figure out why the test_reindexing integration test failed. Can you give some instructions on how to proceed?

**Comment by gfyoung** [comment]
@BBQHarvey : Don't worry about it.  #17355 will take care of it.

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -129,7 +129,7 @@ Other Enhancements
 - `read_*` methods can now infer compression from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`).
 - :func:`pd.read_sas()` now recognizes much more of the most frequently used date (datetime) formats in SAS7BDAT files (:issue:`15871`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: move to 0.21.1

**Review by jreback — CHANGES_REQUESTED** [review]
pls rebase

**Comment by jreback** [comment]
closing as stale

**Pull request closed without merging**